modules/homa-lite/src/lib.rs

```diff
@@ -293,29 +360,504 @@ pub mod module {
+				if liquid_remaining.is_zero() || redeem_requests_limit_remaining.is_zero() {
+					break;
+				}
+				Self::match_mint_with_redeem_request(
+					minter,
+					&redeemer,
+					request_amount,
+					extra_fee,
+					&mut liquid_remaining,
+					&mut new_balances,
+				)?;
+				redeem_requests_limit_remaining -= 1;
+			}
+
+			// Update storage to the new balances. Remove Redeem requests that have been filled.
+			Self::update_redeem_requests(&new_balances);
+
+			// If significant balance is left over, the remaining liquid currencies are minted through XCM.
+			let mut staking_remaining = Self::convert_liquid_to_staking(liquid_remaining)?;
+			if staking_remaining > T::MinimumMintThreshold::get().saturating_add(T::MintFee::get()) {
+				// Calculate how much Liquid currency is to be minted.
+				// liquid_to_mint = convert_to_liquid( (staked_amount - MintFee) * (1 - MaxRewardPerEra) )
+				let mut liquid_to_mint = staking_remaining
+					.checked_sub(T::MintFee::get())
+					.expect("Mint amount is ensured to be greater than T::MintFee; qed");
+				liquid_to_mint = (Permill::one().saturating_sub(T::MaxRewardPerEra::get())).mul(liquid_to_mint);
+				liquid_to_mint = Self::convert_staking_to_liquid(liquid_to_mint)?;
+
+				// Ensure the total amount staked doesn't exceed the cap.
+				let new_total_staking_currency = Self::total_staking_currency()
+					.checked_add(staking_remaining)
+					.ok_or(ArithmeticError::Overflow)?;
+				ensure!(
+					new_total_staking_currency <= Self::staking_currency_mint_cap(),
+					Error::<T>::ExceededStakingCurrencyMintCap
+				);
+
+				TotalStakingCurrency::<T>::put(new_total_staking_currency);
+
+				// All checks pass. Proceed with Xcm transfer.
+				T::XcmTransfer::transfer(
+					minter.clone(),
+					staking_currency,
+					staking_remaining,
+					T::SovereignSubAccountLocation::get(),
+					Self::xcm_dest_weight(),
+				)?;
+				T::Currency::deposit(T::LiquidCurrencyId::get(), minter, liquid_to_mint)?;
+
+				staking_remaining = Balance::zero();
+				liquid_remaining = liquid_remaining
+					.checked_sub(liquid_to_mint)
+					.expect("Liquid amount cannot be higher after fees are deducted; qed");
+			}
+
+			let actual_staked = amount
+				.checked_sub(staking_remaining)
+				.expect("Staking remaining cannot be more than the original; qed");
+			let actual_liquid = total_liquid_to_mint
+				.checked_sub(liquid_remaining)
+				.expect("Liquid remaining cannot be more than the original; qed");
+
+			Self::deposit_event(Event::<T>::Minted(minter.clone(), actual_staked, actual_liquid));
+
+			Ok(())
+		}
+
+		#[transactional]
+		fn process_scheduled_unbond(staking_amount: Balance) -> DispatchResult {
+			let msg = Self::construct_xcm_unreserve_message(T::ParachainAccount::get(), staking_amount);
+
+			let res = pallet_xcm::Pallet::<T>::send_xcm(Here, Parent.into(), msg);
+			//(origin, MultiLocation::X1(Junction::Parent.into()), msg.into());
+			if res.is_ok() {
```

Comment:
missed `else`.  Should check the error here, then the `if` block can be removed.